diff --git a/dist/popover-fn.js b/dist/popover-fn.js
index 34cad556cad4c8bcece443211feea2d24cf493dc..d0f1f74757ab32d02c5592e9683c89f7f367205b 100644
--- a/dist/popover-fn.js
+++ b/dist/popover-fn.js
@@ -489,7 +489,7 @@ var ShadowRoot2 = globalThis.ShadowRoot || function() {
 };
 var DEFAULT_LAYER_NAME = "popover-polyfill";
 function isSupported() {
-  return typeof HTMLElement !== "undefined" && typeof HTMLElement.prototype === "object" && "popover" in HTMLElement.prototype;
+  return typeof HTMLElement !== "undefined" && typeof HTMLElement.prototype === "object" && "popover" in HTMLElement.prototype && "showPopover" in HTMLElement.prototype;
 }
 function isHintSupported() {
   const el = document.createElement("div");
@@ -517,7 +517,7 @@ function hasLayerSupport() {
 function getStyles(layerName) {
   var _a;
   const useLayer = hasLayerSupport();
-  const layerNameEscaped = (layerName ?? ((_a = window.POPOVER_POLYFILL_OPTIONS) == null ? void 0 : _a.layerName) ?? DEFAULT_LAYER_NAME).split(".").map(CSS.escape).join(".");
+  const layerNameEscaped = (layerName ?? ((_a = window.POPOVER_POLYFILL_OPTIONS) == null ? void 0 : _a.layerName) ?? DEFAULT_LAYER_NAME).split(".").map((c) => CSS.escape(c)).join(".");
   return `
 ${useLayer ? `@layer ${layerNameEscaped} {` : ""}
   :where([popover]) {
@@ -593,7 +593,7 @@ function injectStyles(root, layerName) {
       popoverStyleSheet = false;
     }
   }
-  if (popoverStyleSheet === false) {
+  if (popoverStyleSheet === false || !root.adoptedStyleSheets) {
     const sheet = document.createElement("style");
     sheet.textContent = styles;
     if (root instanceof Document) {
@@ -651,6 +651,7 @@ function apply(opts) {
     showPopover: {
       enumerable: true,
       configurable: true,
+      writable: true,
       // eslint-disable-next-line @typescript-eslint/no-unused-vars
       value(options = {}) {
         showPopover(this);
@@ -659,6 +660,7 @@ function apply(opts) {
     hidePopover: {
       enumerable: true,
       configurable: true,
+      writable: true,
       value() {
         hidePopover(this, true, true);
       }
@@ -666,6 +668,7 @@ function apply(opts) {
     togglePopover: {
       enumerable: true,
       configurable: true,
+      writable: true,
       value(options = {}) {
         if (typeof options === "boolean") {
           options = { force: options };
@@ -803,7 +806,8 @@ function apply(opts) {
     const key = event.key;
     const target = event.target;
     if (!event.defaultPrevented && target && (key === "Escape" || key === "Esc")) {
-      hideAllPopoversUntil(target.ownerDocument, true, true);
+      const owner = target.ownerDocument || target;
+      hideAllPopoversUntil(owner, true, true);
     }
   };
   const addEventListeners = (root) => {
